From 7898c737aed13667c22830d2379122e15cdf96b9 Mon Sep 17 00:00:00 2001
From: Jiri Slaby <jslaby@suse.cz>
Date: Tue, 19 Apr 2022 08:30:09 +0200
Subject: [PATCH 649/921] [Backport] scripts: dummy-tools, add pahole

mainline inclusion
from mainline-v5.18-rc1
commit a90bb65ae2168c8b36e53f82dc3cb35c6cff4f1e
category: bugfix
bugzilla: 188608

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a90bb65ae2168c8b36e53f82dc3cb35c6cff4f1e

------------------------------------------------------

CONFIG_PAHOLE_VERSION is a part of a config since the commit below. And
when multiple people update the config, this value constantly changes.
Even if they use dummy scripts.

To fix this, add a pahole dummy script returning v99.99. (This is
translated into 9999 later in the process.)

Thereafter, this script can be invoked easily for example as:
make PAHOLE=scripts/dummy-tools/pahole oldconfig

Fixes: 613fe1692377 (kbuild: Add CONFIG_PAHOLE_VERSION)
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
Signed-off-by: Yang Jihong <yangjihong1@huawei.com>
---
 scripts/dummy-tools/pahole | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100755 scripts/dummy-tools/pahole

diff --git a/scripts/dummy-tools/pahole b/scripts/dummy-tools/pahole
new file mode 100755
index 000000000000..53501a36fa71
--- /dev/null
+++ b/scripts/dummy-tools/pahole
@@ -0,0 +1,4 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0-only
+
+echo v99.99
-- 
2.17.1

